src/pages/api/sanity/lessons/create.ts

```diff
@@ -75,6 +67,22 @@ const sanityIdForDocumentType = async (
   return `${documentType}-${id}`
 }
 
+const courseSchema = z.object({
+  title: z.string(),
+  collaboratorId: z.string().optional(),
+  topicIds: z.string().array(),
+})
+export type CourseData = z.infer<typeof courseSchema>
```

Comment:
First, define a `courseSchema` that we can use to parse incoming request data in the main function.

Second, `infer` a type from that schema that can replace the `CourseData` type we are importing. Very little code needs to change since this type is one-for-one with the other type.

Reply:
I haven't seen Zod before. It looks a lot like Yup to me https://github.com/jquense/yup